pkg/operators/wasm/maps.go

```
@@ -236,7 +255,31 @@ func (i *wasmOperatorInstance) mapUpdate(ctx context.Context, m wapi.Module, sta
 		return
 	}
 
-	err = ebpfMap.Update(key, value, ebpfFlags)
+	if getFD {
+		if !mapAcceptsFD(ebpfMap) {
+			i.logger.Warnf("mapUpdate: map type %v does not accept FD, expecting one in: %v", ebpfMap.Type(), mapTypesAcceptingFD)
+			stack[0] = 1
+			return
+		}
+
+		targetMapHandle := binary.NativeEndian.Uint32(value)
+		targetMap, ok := getHandle[*ebpf.Map](i, targetMapHandle)
+		if !ok {
+			stack[0] = 1
+			return
+		}
+
+		err = ebpfMap.Update(key, uint32(targetMap.FD()), ebpfFlags)
+	} else {
+		if mapAcceptsFD(ebpfMap) {
+			i.logger.Warnf("mapUpdate: map type %v accepts FD, expecting any map except one in: %v", ebpfMap.Type(), mapTypesAcceptingFD)
+			stack[0] = 1
+			return
+		}
+
+		err = ebpfMap.Update(key, value, ebpfFlags)
+	}
```

At this point I'm really wondering if we need the getFD flag. Can't we just imply it from the map type? If trying to update a "mapAcceptsFD" map, then treat the value as a handle. I know I suggested the flag before, but if we're going to implement it in this way, I think we can avoid it. 

Is there anything I am missing?

---

Removing the getFD flag is fine for me.

There are other map kinds that take a fd, such as BPF_MAP_TYPE_PERF_EVENT_ARRAY, BPF_MAP_TYPE_PROG_ARRAY, BPF_MAP_TYPE_CGROUP_ARRAY, etc. The name of the function `mapAcceptsFD` does not reflect that.

But those don't accept the fd of a bpf map, but a fd of different types.

We don't have any wasm function to get a fd from a cgroup, so we don't support BPF_MAP_TYPE_CGROUP_ARRAY. We need to return an error if a wasm program attemps to access it, regardless of the getFD flag.

The function `mapAcceptsFD` needs to be replaced by something that tells if the map:
- Is unsupported (e.g. BPF_MAP_TYPE_CGROUP_ARRAY), so it can't be used from wasm
- Is supported and takes the fd of a bpf map
- Is supported and takes a regular buffer

It seems to me it cannot just be a function returning a `bool`.

Maps of unknown types (from newer Linux releases) should be unsupported by default.